routers/web/org/times.go

```diff
@@ -0,0 +1,218 @@
+// Copyright 2014 The Gogs Authors. All rights reserved.
+// Copyright 2022 The Gitea Authors.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package org
+
+import (
+	"net/http"
+	"time"
+
+	"code.gitea.io/gitea/models/db"
+	"code.gitea.io/gitea/modules/base"
+	"code.gitea.io/gitea/modules/context"
+	"code.gitea.io/gitea/modules/setting"
+
+	"xorm.io/builder"
+)
+
+const (
+	tplByRepos      base.TplName = "org/times/times_by_repos"
+	tplByMembers    base.TplName = "org/times/times_by_members"
+	tplByMilestones base.TplName = "org/times/times_by_milestones"
+)
+
+// resultTimesByRepos is a struct for DB query results
+type resultTimesByRepos struct {
+	Name    string
+	SumTime int64
+}
+
+// resultTimesByMilestones is a struct for DB query results
+type resultTimesByMilestones struct {
+	RepoName     string
+	Name         string
+	ID           string
+	SumTime      int64
+	HideRepoName bool
+}
+
+// resultTimesByMembers is a struct for DB query results
+type resultTimesByMembers struct {
+	Name    string
+	SumTime int64
+}
+
+// parseTimes contains functionality that is required in all these functions,
+// like parsing the date from the request, setting default dates, etc.
+func parseTimes(ctx *context.Context) (unixfrom, unixto int64, err error) {
+	// View variables
+	ctx.Data["PageIsOrgTimes"] = true
+	ctx.Data["AppSubURL"] = setting.AppSubURL
+
+	// Time range from request, if any
+	from := ctx.FormString("from")
+	to := ctx.FormString("to")
+	// Defaults for "from" and "to" dates, if not in request
+	if from == "" {
+		// DEFAULT of "from": start of current month
+		from = time.Now().Format("2006-01") + "-01"
+	}
+	if to == "" {
+		// DEFAULT of "to": today
+		to = time.Now().Format("2006-01-02")
+	}
+
+	// Prepare Form values
+	ctx.Data["RangeFrom"] = from
+	ctx.Data["RangeTo"] = to
+
+	// Prepare unix time values for SQL
+	from2, err := time.Parse("2006-01-02", from)
+	if err != nil {
+		ctx.ServerError("time.Parse", err)
+	}
+	unixfrom = from2.Unix()
+	to2, err := time.Parse("2006-01-02", to)
+	if err != nil {
+		ctx.ServerError("time.Parse", err)
+	}
+	// Humans expect that we include the ending day too
+	unixto = to2.Add(1440*time.Minute - 1*time.Second).Unix()
+	return unixfrom, unixto, err
+}
+
+// TimesByRepos renders worktime by repositories.
+func TimesByRepos(ctx *context.Context) {
+	// Run common functionality
+	unixfrom, unixto, err := parseTimes(ctx)
+	if err != nil {
+		return
+	}
+
+	// Set submenu tab
+	ctx.Data["TabIsByRepos"] = true
+
+	results, err := getTimesByRepos(unixfrom, unixto, ctx.Org.Organization.ID)
+	if err != nil {
+		ctx.ServerError("getTimesByRepos", err)
+		return
+	}
+	ctx.Data["results"] = results
+
+	// Reply with view
+	ctx.HTML(http.StatusOK, tplByRepos)
+}
+
+// getTimesByRepos fetches data from DB to serve TimesByRepos.
+func getTimesByRepos(unixfrom, unixto, orgid int64) (results []resultTimesByRepos, err error) {
```

Comment:
These kind of functions lives in `models/` directory. `routers/` is just connecting all the functions and do the nice things with the data.